client/v2_2/docker_image_.py

```diff
@@ -453,6 +453,12 @@ def uncompressed_blob(self, digest):
     return self._content(self._blob_names[digest],  # pytype: disable=none-attr
                          memoize=False)
 
+  def layer_blob(self, index):
+    """Returns the uncompressed blob for the layer at the given index."""
+    if index < len(self._layers):
```

Comment:
`layer_blob` is ambiguous about compression, and I don't like the use of index vs. diff-id :-/

Reply:
I can change the name to uncompressed_layer_blob, that's not a problem.

I could not get a simple implementation for a diff-id based solution. I would have to extract the diff-ids from the config, preserve and expose their ordering, _compute_ the diff ids of all the layers, and then maintain a map from diff-id to layer filename. The consumer will have to iterate over the diff ids when in our case, we do not care about it.

The actual code for this may be small, but it is doing things that are not needed for this use case. If you envision the diff ids being useful in the future, then I can make another attempt.

Reply:
Ah, I won't have to compute the diff-ids, but the rest of the complexity will still be there.

If you want, I can prepare an option with diff ids and then you can choose based on what you like. 😄

Reply:
I updated the branch with a diff_id based solution. I can go ahead with whichever you like better.

Reply:
I definitely prefer the `diff_id` based solution, and I think this is pretty close.

For this repo, the source of truth is actually in our mono-repo (with pytype annotations and tests) and we export this stripped version.  Let me get a change going there and I'll export a PR we can continue to iterate on.